azkaban-common/src/main/java/azkaban/executor/ExecutableNode.java

```diff
@@ -132,7 +132,7 @@ public void setType(String type) {
     this.type = type;
   }
 
-  public void setStatus(Status status) {
+  public synchronized void setStatus(Status status) {
```

Comment:
Nice one 👍 Actually I tried to revert this change and run some tests but couldn't get them fail any more (I didn't add `volatile` yet). I would take a bit of more effort to figure out if there's a real need for any of this, but making it `volatile` any way sounds good to me.